pkg/controller/nexus/resource/deployment/manager.go

```diff
@@ -0,0 +1,123 @@
+//     Copyright 2020 Nexus Operator and/or its authors
+//
+//     This file is part of Nexus Operator.
+//
+//     Nexus Operator is free software: you can redistribute it and/or modify
+//     it under the terms of the GNU General Public License as published by
+//     the Free Software Foundation, either version 3 of the License, or
+//     (at your option) any later version.
+//
+//     Nexus Operator is distributed in the hope that it will be useful,
+//     but WITHOUT ANY WARRANTY; without even the implied warranty of
+//     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//     GNU General Public License for more details.
+//
+//     You should have received a copy of the GNU General Public License
+//     along with Nexus Operator.  If not, see <https://www.gnu.org/licenses/>.
+
+package deployment
+
+import (
+	ctx "context"
+	"fmt"
+	"github.com/RHsyseng/operator-utils/pkg/resource"
+	"github.com/m88i/nexus-operator/pkg/apis/apps/v1alpha1"
+	"github.com/m88i/nexus-operator/pkg/logger"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const mgrNotInit = "the manager has not been initialized"
+
+var log = logger.GetLogger("deployment_manager")
+
+// Manager is responsible for creating deployment-related resources, fetching deployed ones and comparing them
+type Manager struct {
```

Comment:
Do we need to export this `struct`?
You may keep it private and create an `interface` instead. So the `NewManager` function would return the `manager` struct, that would implement the interface `Manager`. (or if `Manager` already implements the interface you created in another package, just keep it private since it's a structure that will only be used/created in here, and the builder `NewManager` is already handling it)

Reply:
Oh, nicely spotted! I definitely agree, thanks!